src/hooks/useVenueUserStatuses.ts

```diff
@@ -1,30 +1,46 @@
-import {
-  ONLINE_USER_STATUS,
-  DEFAULT_SHOW_USER_STATUSES,
-  USER_STATUSES,
-} from "settings";
+import { useCallback } from "react";
+
+import { updateUserOnlineStatus } from "api/profile";
+
+import { DEFAULT_USER_STATUS, DEFAULT_SHOW_USER_STATUSES } from "settings";
 
 import { User } from "types/User";
 
 import { WithId } from "utils/id";
 
 import { useSovereignVenue } from "./useSovereignVenue";
+import { useUser } from "./useUser";
 
 export const useVenueUserStatuses = (venueId?: string, user?: WithId<User>) => {
   const { sovereignVenue } = useSovereignVenue({ venueId });
+  const { userId, profile } = useUser();
+
+  const changeUserStatus = useCallback(
+    (newStatus?: string) => {
+      if (!userId) return;
+
+      updateUserOnlineStatus({
+        status: newStatus,
+        userId,
+      });
+    },
+    [userId]
+  );
 
   const venueStatuses = sovereignVenue?.userStatuses ?? [];
 
-  const statuses = [...USER_STATUSES, ...venueStatuses];
-
-  const userStatus = statuses.find(
+  const userStatus = venueStatuses.find(
     (userStatus) => userStatus.status === user?.status
   );
 
   return {
+    changeUserStatus,
     venueUserStatuses: venueStatuses,
     isStatusEnabledForVenue:
       sovereignVenue?.showUserStatus ?? DEFAULT_SHOW_USER_STATUSES,
-    userStatus: userStatus ?? ONLINE_USER_STATUS,
+    userStatus: userStatus ?? {
+      status: profile?.status ?? "",
+      color: DEFAULT_USER_STATUS.color,
```

Comment:
seems like we need to use here `venueStatuses[0].color` as well as per discussion: https://github.com/sparkletown/sparkle/pull/1772#discussion_r667032778.
```suggestion
      color: venueStatuses[0]?.color ?? DEFAULT_USER_STATUS.color,
```

Reply:
Correct. I just pushed the change in https://github.com/sparkletown/sparkle/pull/1772/commits/f61564ac56ee8bec1ee7997723fd34c47c21636b

Reply:
@denisdimitrov oo, looking at your changes. Maybe it would be better to do something like this?
```
userStatus: userStatus ?? venueStatuses[0] ?? DEFAULT_USER_STATUS;
```

Reply:
Just saw the comment, looks like a valid solution. We might actually be able to remove that useEffect that checks if the profile exists.